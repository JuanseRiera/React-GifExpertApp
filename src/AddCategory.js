import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setcategories }) => {
  const [inputValue, setinputValue] = useState("");

  const manageChange = (e) => {
    setinputValue(e.target.value.toUpperCase());
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setcategories((categories) => {
        return [inputValue.toUpperCase(), ...categories];
      });
      setinputValue("");
    }
  };

  return (
    <>
      <form onSubmit={onSubmit}>
        <input
          type="text"
          id="add"
          autoComplete="off"
          placeholder="Buscar GIF..."
          value={inputValue}
          onChange={manageChange}
        />
      </form>
    </>
  );
};

AddCategory.propTypes = {
  setcategories: PropTypes.func.isRequired,
};
