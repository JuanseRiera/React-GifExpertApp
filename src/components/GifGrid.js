import React from "react";
import PropTypes from "prop-types";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  let { data, loading } = useFetchGifs(category);

  const htmlToRender = () => {
    if (!loading && data && data.length > 0) {
      return (
        <div className="cardContainer">
          {data?.map((img) => {
            return <GifGridItem key={img.id} {...img} />;
          })}
        </div>
      );
    }
    if (!loading && (!data || data.length === 0)) {
      return (
        <>
          <p className="nombreCategoria animate__animated animate__headShake">
            No se encontraron elementos para la busqueda realizada.
          </p>
        </>
      );
    }
  };
  return (
    <>
      <h3 className="nombreCategoria animate__animated animate__flipInY">
        {category}
      </h3>
      {loading && (
        <p className="nombreCategoria animate__animated animate__flash">
          Cargando...
        </p>
      )}
      {htmlToRender()}
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
