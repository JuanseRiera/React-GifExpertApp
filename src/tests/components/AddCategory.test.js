import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { AddCategory } from "../../components/AddCategory";

describe("Pruebas para el componente addCategory", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setcategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setcategories={setCategories} />);
  });

  test("Debe de mostrarse correctamente", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("Deberia cambiar los valores del input", () => {
    const value = "hola mundo";
    const input = wrapper.find("input");

    input.simulate("change", { target: { value } });
    expect(wrapper.find("input").prop("value")).toBe(value.toUpperCase());
  });

  test("No se deberia ejecutar la funcion set categories", () => {
    let form = wrapper.find("form");
    form.simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Deberia llamar al setCategories y limpiar la caja de texto", () => {
    const input = wrapper.find("input");
    input.simulate("change", { target: { value: "What up" } });

    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    expect(input.prop("value")).toBe("");
  });
});
