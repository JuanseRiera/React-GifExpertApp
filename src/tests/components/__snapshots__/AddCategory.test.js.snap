// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente addCategory Debe de mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      id="add"
      onChange={[Function]}
      placeholder="Buscar GIF..."
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
